fireapp/lib/presentation/supervisor_shifts/supervisor_shifts_view_model.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:fireapp/domain/repository/shifts_repository.dart';
+import 'package:flutter/material.dart';
+import 'package:injectable/injectable.dart';
+
+import '../../global/di.dart';
+
+@Injectable()
+class SupervisorShiftsViewModel {
+  final ShiftsRepository shiftsRepository;
+
+  SupervisorShiftsViewModel(this.shiftsRepository);
+
+  Future<void> optimiseShifts() async {
+    try {
+      await shiftsRepository.optimiseShifts();
```

Comment:
Add some kind of progress indicator for the network request (not the optimization process)

Reply:
There's not much to show atm. I gave it a try but not sure if it's good enough.

Reply:
What you've done is totally adequate, I'm happy with what you did. If something more is needed in the future, the future dev can add it.